### ISSUE TITLE
Updating multiple database entries at once

### ISSUE THREAD

**Issue by sahanDissanayake** [balderdashy/sails#2404]
Hi,

This is my function to update multiple database values at once

``` javascript
Menuitem.update([{id: 1},{position: 3}], [{id: 2},{position: 3}]).exec(function(err, updatedRecords) {
                if (err) {
                    return res.send({message: 'Could not update the records', err: err}, 500);
                }

                if (updatedRecords) {
                    return res.send({ records: updatedRecords }, 200);
                } else {
                    return res.notFound('Records not found');
                }
            });
```

Checked the docs at https://github.com/balderdashy/waterline-docs/blob/master/query-methods.md#update-search-criteria--values--callback-

But could not find how to update multiple records at once.
http://stackoverflow.com/questions/26952705/update-multiple-records-at-once-using-sailsjs

**Comment by Clan-Utility** [comment]
According to the [doc](http://sailsjs.org/documentation/reference/waterline-orm/models/update), the 2nd parameter of update() accepts [{}]. What does it mean?

**Comment by mcodev1** [comment]
@sgress454 
I reached this issue while trying to find out how to batch update, so I'm also very curious what that `[{}]` in the docs is referring to. I'm using Mongo as my database and it's throwing an error when I provide an array of objects as the second parameter of `.update()`

**Comment by carpiediem** [comment]
@sgress454 , if your answer is to check the documentation, you should really ensure that it addresses the issue.  As the previous two posts show, the docs seem to indicate that this should work.

**Comment by sgress454** [comment]
@carpiediem That is very fair.  I made the mistake of assuming the docs were correct.  They usually are!  I've updated them in the docs repo, and will see to it that the site is re-deployed soon.

For future reference:

The _find criteria_ (first argument) for `.update()` can either be a dictionary (same kind as you'd use for `.find()`) or, as a shortcut, a primary key value.  To update multiple records by primary key, use an "in" query, e.g. `User.update({ id: {in: [1, 2, 3]} } ...` or the shortcut `User.update({ id: [1, 2, 3]}...`

The _values_ (second argument) for `.update()` _must_ be a dictionary of values to set.  Waterline doesn't support using an array as the second argument to set different values for different records.  

The [Sails 1.0 docs](http://next.sailsjs.com/documentation/reference/waterline-orm/models/update#?parameters) are correct, at least 🙃
